### PR TITLE
Colorcontrol Default Implementation

### PR DESCRIPTION
**Pull request by Apollon77**
This PR adds:
* a full blown feature complete default implementation for the ColorControl cluster
* it supports managed transition mode like LevelControl
* it tries to convert colors between color modes best effort based
* Adds Levelcontrol dependency and optimizations/fixes in levelControl implementation

### CODE REVIEW

File: models/src/local/ColorControlOverrides.ts [hunk 1]
```diff
@@ -47,6 +87,60 @@ LocalMatter.children.push({
         },
 
         // Spec states the values of this bitmap are the same as the feature map
-        { tag: "attribute", id: 0x400a, name: "ColorCapabilities", type: "FeatureMap" },
```
Comment (lauckhart) [inline models/src/local/ColorControlOverrides.ts:50]: Why'd you have to do this?  Was there an issue using the FeatureMap enum?

File: models/src/local/ColorControlOverrides.ts [hunk 2]
```diff
@@ -47,6 +87,60 @@ LocalMatter.children.push({
...
+                },
+                {
+                    tag: "field",
+                    name: "ColorTemperature",
+                    constraint: "4",
+                },
+            ],
+        },
+
+        // Set the correct type of ModeMode because just in the description
```
Comment (lauckhart) [inline models/src/local/ColorControlOverrides.ts:124]: "MoveMode"

File: packages/matter.js/src/behavior/definitions/color-control/ColorControlServer.ts [hunk 3]
```diff
@@ -4,11 +4,2232 @@
...
+            colorTemperatureMaximumMireds: number,
+        ): MaybePromise;
+        switchColorMode(oldMode: ColorControl.ColorMode, newMode: ColorControl.ColorMode): MaybePromise;
+        setColorMode(mode: ColorControl.ColorMode): MaybePromise;
+        setEnhancedColorMode(mode: ColorControl.EnhancedColorMode): MaybePromise;
+        syncColorTemperatureWithLevelLogic(level: number): MaybePromise;
+    };
+}
+
+// We had turned on some more features to provide da default implementation, but export the cluster with default
```
Comment (lauckhart) [inline packages/matter.js/src/behavior/definitions/color-control/ColorControlServer.ts:2233]: "da" = "a"?

**Review by lauckhart — APPROVED** [review]

File: models/src/local/ColorControlOverrides.ts (hunk 1 above)
Comment (Apollon77) [reply models/src/local/ColorControlOverrides.ts:50]: Yes. I could not assigning to this.features because is a bit different in structure type wise.

**Pull request merged**